app/server/api/vendor_payout_api.py

```diff
@@ -65,58 +67,74 @@ def post(self):
         writer = csv.writer(output)
 
         writer.writerow([
-            'ID',
+            'Vendor Account ID',
             'Phone',
-            'ContactName',
-            'Current Balance',
-            'Total Sent',
+            'ContactName', 
+            'Current Balance', 
+            'Total Sent', 
             'Total Received',
-            'Approved',
-            'Beneficiary',
+            'Approved', 
+            'Beneficiary', 
             'Vendor',
             'InvoiceDate',
             'DueDate',
-            'Transaction ID',
+            'Transfer ID',
             'UnitAmount',
             'Payment Has Been Made',
             'Bank Payment Date',
         ])
         for v in vendors:
-            transfer = make_withdrawal_transfer(
-                Decimal(v._balance_wei or 0) / Decimal(1e16),
-                token=v.token,
-                send_user=v.primary_user,
-                sender_transfer_account=v,
-                transfer_mode=TransferModeEnum.INTERNAL,
-                require_sender_approved=False,
-                automatically_resolve_complete=False,
-            )
-
-            db.session.flush()
+            if relist_existing:
+                withdrawals = (CreditTransfer.query
+                               .filter(CreditTransfer.sender_transfer_account_id == v.id)
+                               .filter(CreditTransfer.transfer_status == TransferStatusEnum.PENDING).all())
+            else:
+                withdrawals = []
+
+            withdrawal_amount = Decimal(v._balance_wei or 0) / Decimal(1e16)
+            if withdrawal_amount > 0:
+
+                transfer = make_withdrawal_transfer(
+                    withdrawal_amount,
+                    token=v.token,
+                    send_user=v.primary_user,
+                    sender_transfer_account=v,
+                    transfer_mode=TransferModeEnum.INTERNAL,
+                    require_sender_approved=False,
+                    automatically_resolve_complete=False,
+                )
+
+                db.session.flush()
+
+                withdrawals.append(transfer)
+
+            for w in withdrawals:
+                writer.writerow([
+                    v.id,
+                    v.primary_user.phone,
+                    v.primary_user.first_name + ' ' + v.primary_user.last_name,
```

Comment:
I get this error when first or last name is null - maybe worth checking it exists or doing `str()` to ensure a 500 isn't accidently returned
```
	<title>TypeError: can only concatenate str (not &quot;NoneType&quot;) to str // Werkzeug Debugger</title>
```

Reply:
ohhh great catch. will add